db/db_compaction_test.cc

```diff
@@ -9157,63 +9165,11 @@ TEST_F(DBCompactionTest, TurnOnLevelCompactionDynamicLevelBytes) {
   // newly flushed file is also pushed down
   options.level_compaction_dynamic_level_bytes = true;
   Reopen(options);
-  ASSERT_EQ("0,0,0,1,1,2", FilesPerLevel());
-  verify_db();
-}
-
-TEST_F(DBCompactionTest, TurnOnLevelCompactionDynamicLevelBytesIngestBehind) {
-  Options options = CurrentOptions();
-  options.compaction_style = kCompactionStyleLevel;
-  options.allow_ingest_behind = true;
-  options.level_compaction_dynamic_level_bytes = false;
-  options.num_levels = 6;
-  options.compression = kNoCompression;
-  DestroyAndReopen(options);
-
-  // put files in L0, L1 and L2
-  WriteOptions write_opts;
-  ASSERT_OK(db_->Put(write_opts, Key(1), "val1"));
-  ASSERT_OK(Flush());
-  MoveFilesToLevel(2);
-  ASSERT_OK(db_->Put(write_opts, Key(2), "val2"));
-  ASSERT_OK(Flush());
-  MoveFilesToLevel(2);
-  ASSERT_OK(db_->Put(write_opts, Key(1), "new_val1"));
-  ASSERT_OK(Flush());
-  MoveFilesToLevel(1);
-  ASSERT_OK(db_->Put(write_opts, Key(3), "val3"));
-  ASSERT_OK(Flush());
-  ASSERT_EQ("1,1,2", FilesPerLevel());
-  auto verify_db = [&]() {
-    ASSERT_EQ(Get(Key(1)), "new_val1");
-    ASSERT_EQ(Get(Key(2)), "val2");
-    ASSERT_EQ(Get(Key(3)), "val3");
-  };
-  verify_db();
-
-  options.level_compaction_dynamic_level_bytes = true;
-  Reopen(options);
-  // note that last level (L6) should be empty
-  ASSERT_EQ("1,0,0,1,2", FilesPerLevel());
-  verify_db();
-
-  // turning the options on and off should both be safe
-  options.level_compaction_dynamic_level_bytes = false;
-  Reopen(options);
-  MoveFilesToLevel(1);
-  ASSERT_EQ("0,1,0,1,2", FilesPerLevel());
-  verify_db();
-
-  // newly flushed file is also pushed down
-  options.level_compaction_dynamic_level_bytes = true;
-  Reopen(options);
-  ASSERT_EQ("0,0,1,1,2", FilesPerLevel());
-  verify_db();
-
-  // files will be pushed down to last level (L6)
-  options.allow_ingest_behind = false;
-  Reopen(options);
-  ASSERT_EQ("0,0,0,1,1,2", FilesPerLevel());
```

Comment:
I did find it difficult to adapt since there is some unnecessary level in the LSM which can be drained in the background. I could not just disable auto compaction since it would also disallow trivially moving files down the LSM during DB open. However, I eventually found out a [better reason](https://github.com/facebook/rocksdb/pull/11119) to remove the test :) Otherwise, I should probably just verify the number of last level as you suggested.

Reply:
Yeah, IIRC there was a comment elsewhere in this PR referring to the better reason - but allow_ingest_behind feels so close to working with leveled compaction that we might as well start testing it when convenient and when there's known risks. Maybe the test can say support is not officially released since on rare occasions people infer what's supported by looking at test expectations

Reply:
> just verify the number of files in the bottom level (L5 since num_levels=6) is zero before the final Reopen(), and nonzero after it.

I tried fixing this, but realize this does not work (without adding support for level compaction + ingest behind) as the second last level can become unnecessary and compacted down even when the last level is reserve for ingest behind.  Making it work requires more changes that almost feels like supporting allow_ingest_behind with level compaction in this PR. I filed a task and plan to follow up with a separate PR for official support.